### PR TITLE
Add support to run remote template

### PR DESCRIPTION
**Pull request by parrasajad**
* Add remote-template-domain config only flag to specify allowed domain list to load remote templates from

## Checklist

- [x] Pull request is created against the [dev](https://github.com/projectdiscovery/nuclei/tree/dev) branch
- [x] All checks passed (lint, unit/integration/regression tests etc.) with my changes
- [x] I have added tests that prove my fix is effective or that my feature works
- [ ] I have added necessary documentation (if appropriate)

### CODE REVIEW

**Review by ehsandeep — CHANGES_REQUESTED** [review]
Requires `go mod` update to fetch changes from goflags

**Review by forgedhallpass — CHANGES_REQUESTED** [review]
Since this is a sensitive feature that can lead to potential vulnerabilities if not implemented correctly, it is important to have extensive tests with multiple scenarios, especially negative.

File: v2/pkg/catalog/loader/remote_loader.go [hunk 1]
```diff
@@ -55,7 +56,27 @@ func getRemoteTemplatesAndWorkflows(templateURLs []string, workflowURLs []string
 	return remoteTemplateList, remoteWorkFlowList, err
 }
 
-func getRemoteContent(URL string, w chan<- RemoteContentError, contentType ContentType) {
+func getRemoteContent(URL string, remoteTemplateDomainList []string, w chan<- RemoteContentError, contentType ContentType) {
+	if strings.HasPrefix(URL, "http") && (strings.HasSuffix(URL, ".yaml") || strings.HasSuffix(URL, ".yml")) {
+		parsed, err := url.Parse(URL)
```
Comment (forgedhallpass) [inline v2/pkg/catalog/loader/remote_loader.go:61]: please state exactly what a variable holds. It's a `parsedURL`, not simply just "parsed"

File: v2/pkg/catalog/loader/remote_loader.go [hunk 2]
```diff
@@ -55,7 +56,27 @@ func getRemoteTemplatesAndWorkflows(templateURLs []string, workflowURLs []string
 	return remoteTemplateList, remoteWorkFlowList, err
 }
 
-func getRemoteContent(URL string, w chan<- RemoteContentError, contentType ContentType) {
+func getRemoteContent(URL string, remoteTemplateDomainList []string, w chan<- RemoteContentError, contentType ContentType) {
```
Comment (forgedhallpass) [inline v2/pkg/catalog/loader/remote_loader.go:59]: Please rename the `w` to something more meaningful

File: v2/pkg/templates/compile.go [hunk 3]
```diff
@@ -105,6 +100,30 @@ func Parse(filePath string, preprocessor Preprocessor, options protocols.Execute
 	parsedTemplatesCache.Store(filePath, template, err)
 	return template, nil
 }
+func readFromTemplatePath(templatePath string) (data []byte, err error) {
```
Comment (forgedhallpass) [inline v2/pkg/templates/compile.go:103]: this method seems to be duplicated...please extract and reuse

**Review by forgedhallpass — CHANGES_REQUESTED** [review]

**Review by forgedhallpass — APPROVED** [review]

**Review by ehsandeep — CHANGES_REQUESTED** [review]
Currently `RemoteTemplateDomainList` validation cheks only when loading single remote template with YAML extension and not when loading remote template with list input, we need to enforce the validation for 2nd input as well.

**Review by ehsandeep — APPROVED** [review]

**Pull request merged**